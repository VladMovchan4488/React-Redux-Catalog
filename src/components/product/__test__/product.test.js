import React from "react";
import Product from "../product";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

it("Snapshot my Product", () => {
  const params = {
    img:
      "https://images-na.ssl-images-amazon.com/images/I/71Y4Ogr8BNL._SX679_.jpg",
    name: "A3 Magnetic Dry",
    price: 15.79
  };

  const wrapper = shallow(
    <Product img={params.img} name={params.name} price={params.price} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
