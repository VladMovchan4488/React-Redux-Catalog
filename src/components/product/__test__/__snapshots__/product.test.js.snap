// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Styled(Card)>
  <Styled(CardHeader)>
    <Styled(CardImg)
      src="https://images-na.ssl-images-amazon.com/images/I/71Y4Ogr8BNL._SX679_.jpg"
      variant="top"
    />
  </Styled(CardHeader)>
  <CardBody>
    <CardTitle>
      Название товара:
    </CardTitle>
    <Styled(CardText)>
      A3 Magnetic Dry
    </Styled(CardText)>
  </CardBody>
  <Styled(CardFooter)>
    <Styled(CardText)>
      15.79
      $
    </Styled(CardText)>
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="outline-secondary"
    >
      Купить
    </Button>
  </Styled(CardFooter)>
</Styled(Card)>
`;

exports[`Snapshot my Product 1`] = `
<Styled(Card)>
  <Styled(CardHeader)>
    <Styled(CardImg)
      src="https://images-na.ssl-images-amazon.com/images/I/71Y4Ogr8BNL._SX679_.jpg"
      variant="top"
    />
  </Styled(CardHeader)>
  <CardBody>
    <CardTitle>
      Название товара:
    </CardTitle>
    <Styled(CardText)>
      A3 Magnetic Dry
    </Styled(CardText)>
  </CardBody>
  <Styled(CardFooter)>
    <Styled(CardText)>
      15.79
      $
    </Styled(CardText)>
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="outline-secondary"
    >
      Купить
    </Button>
  </Styled(CardFooter)>
</Styled(Card)>
`;
